backup-restore/uninstall/uninstall-backup-restore.sh

```diff
@@ -0,0 +1,238 @@
+print_heading "Delete any existing guardiancopyrestores CRs"
+oc delete guardiancopyrestores -n ${NAMESPACE} --all
+print_heading "Delete any existing guardianmongoes CRs"
+oc delete guardianmongoes -n ${NAMESPACE} --all
+
+## err_exit "REMOVE THIS"
+
+remove_fsi ()
+{
+   print_heading "Remove any existing fusionserviceinstances"
+   for FBRI in $(oc -n $ISF_NS get fusionserviceinstance --no-headers | awk '{print $1}')
+   do
+           echo "Checking $FBRI"
+           TEMP=$(oc -n $ISF_NS get fusionserviceinstance $FBRI -o json | jq -rc '[(.spec.parameters[]|.name,.value)]|@csv' |grep namespace | cut -d"," -f2| tr -d '"')
+           if [ "$TEMP" == "$NAMESPACE" ]
+            then 
+               print_heading "Update fusionservicedefinition and Remove fusionserviceinstance $FBRI" 
+               FSD=$(oc -n $ISF_NS get fusionserviceinstance $FBRI -o json | jq -rc '.spec.serviceDefinition')
+               oc -n $ISF_NS patch --type json fusionservicedefinition $FSD -p '[{"op": "replace", "path": "/spec/onboarding/serviceOperatorSubscription/triggerCatSrcCreate", "value": false}]'
+               echo "oc delete -n $ISF_NS fusionserviceinstance $FBRI"
+               oc delete -n $ISF_NS fusionserviceinstance $FBRI
+           fi
+   done
+}
+
+remove_fsi
+
+# Subscriptions for dependent operators
+SUBSCRIPTION_NAMES=$(oc -n ${NAMESPACE} get subs --no-headers | cut -d" " -f1)
+
+print_heading "Remove any subscriptions, csvs and operatorgroups for dependent operators in namespace ${NAMESPACE}"
+# Delete the subscription and instance
+for SUBSCRIPTION_NAME in ${SUBSCRIPTION_NAMES[@]}
+do
+    csvName=$(oc get subscription ${SUBSCRIPTION_NAME} -n ${NAMESPACE} -o yaml | grep currentCSV | awk -F' ' '{print $2}')
+    echo "===== Deleting subscription $SUBSCRIPTION_NAME"
+    oc delete subscription ${SUBSCRIPTION_NAME} -n ${NAMESPACE}
+    echo "===== Deleting csv ${csvName}"
+    oc delete clusterserviceversion ${csvName} -n ${NAMESPACE}
+done
+
+# delete the operatorgroup created during install
+print_heading "Delete any existing operatorgroups"
+oc delete operatorgroup -n ${NAMESPACE} --all
+
+print_heading "Delete Redis"
+oc delete redis redis -n ${NAMESPACE} --timeout=60s
+if oc get redis redis -n ${NAMESPACE} >/dev/null 2>&1; then
+   oc patch --type json --patch='[ { "op": "remove", "path": "/metadata/finalizers" } ]' redis redis -n ${NAMESPACE}
+   oc delete redis redis -n ${NAMESPACE}
+fi
+
+if oc get redis redis -n ${NAMESPACE} >/dev/null 2>&1; then
+    echo "Redis instance still exists. You may need to deleted it manually."
+fi
+
+print_heading "Namespace removal started at $(date)"
+echo "Note that this may take a minute or two. If it takes longer than 3 minutes then delete guardian/dataprotection CR instances."
+echo
+
+if ! oc get namespace ${NAMESPACE} >/dev/null 2>&1; then
+    echo "Namespace ${NAMESPACE} was not found"
+  else
+    echo "oc delete namespace ${NAMESPACE}"
+    oc delete namespace ${NAMESPACE}
```

Comment:
if some resources failed to delete above, could this impact the namespace deletion?

Reply:
In theory, yes. Again user will notice hang and investigation what is blocking namespace for getting deleted.